libraries/AP_InertialSensor/AP_InertialSensor.cpp

```diff
@@ -1587,6 +1587,31 @@ void AP_InertialSensor::_save_gyro_calibration()
     }
 }
 
+/*
+  update harmonic notch parameters
+ */
+void AP_InertialSensor::HarmonicNotch::update_params(uint8_t instance, bool converging, float gyro_rate)
+{
+    const float center_freq = calculated_notch_freq_hz[0];
+    if (!is_equal(last_bandwidth_hz, params.bandwidth_hz()) ||
+        !is_equal(last_attenuation_dB, params.attenuation_dB()) ||
+        converging) {
+        filter[instance].init(gyro_rate,
+                              center_freq,
+                              params.bandwidth_hz(),
+                              params.attenuation_dB());
+        last_center_freq_hz = center_freq;
```

Comment:
So this is a subtle change, the "last" parameters have moved from the backend to the frontend. They were in the backend originally because they are per-gyro instance. Also we have locking conventions around backend<->frontend data which we have blurred here. So I think this is probably wrong but will need to think about it some more.

Reply:
the call was always made by the main thread, not the backend bus threads, it being "in the backend" was an illusion already. This just makes it clearer what it really going on

Reply:
though using a semaphore might be a good idea

Reply:
but before these values were per-instance, now they are all shared which I think is not correct - you could have two notches applied to three instances for instance

Reply:
fixed